TESTS/mbed_hal/qspi/qspi_test_utils.cpp

```diff
@@ -0,0 +1,283 @@
+    }
+
+    cmd.address.disabled = (address == QSPI_NONE);
+    if (!cmd.address.disabled) {
+        cmd.address.value = address;
+    }
+
+    cmd.alt.disabled = (alt == QSPI_NONE);
+    if (!cmd.alt.disabled) {
+        cmd.alt.value = alt;
+    }
+}
+
+qspi_command_t* QspiCommand::get()
+{
+    return &cmd;
+}
+
+
+qspi_status_t read_register(uint32_t cmd, uint8_t *buf, uint32_t size, Qspi &q)
+{
+    q.cmd.build(cmd);
+    return qspi_command_transfer(&q.handle, q.cmd.get(), NULL, 0, buf, size);
+}
+
+
+bool flash_wait_for(uint32_t time_us, Qspi &qspi)
+{
+    uint8_t reg[QSPI_STATUS_REGISTER_SIZE];
+    qspi_status_t ret;
+    reg[0] = STATUS_BIT_WIP;
+
+    const ticker_data_t *const ticker = get_us_ticker_data();
+    const uint32_t start = ticker_read(ticker);
+
+    do {
+        ret = read_register(STATUS_REG, reg, QSPI_STATUS_REGISTER_SIZE, qspi);
+    } while (((reg[0] & STATUS_BIT_WIP) != 0) && ((ticker_read(ticker) - start) < (uint32_t)time_us));
+
+    return (((reg[0] & STATUS_BIT_WIP) == 0) && (ret == QSPI_STATUS_OK));
+}
+
+void flash_init(Qspi &qspi)
+{
+    uint8_t status[2] = { 0 };
+    qspi_status_t ret;
+
+    qspi.cmd.build(QSPI_CMD_RDSR);
+    ret = qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, status, 1);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    qspi.cmd.build(QSPI_CMD_RSTEN);
+    ret = qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    qspi.cmd.build(QSPI_CMD_RST);
+    ret = qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    WAIT_FOR(WAIT_MAX_TIME, qspi);
+}
+
+
+qspi_status_t write_enable(Qspi &qspi)
+{
+    uint8_t reg[QSPI_STATUS_REGISTER_SIZE];
+    qspi.cmd.build(QSPI_CMD_WREN);
+
+    if (qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    memset(reg, 0, QSPI_STATUS_REGISTER_SIZE);
+    if (read_register(STATUS_REG, reg, QSPI_STATUS_REGISTER_SIZE, qspi) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+
+    return ((reg[0] & STATUS_BIT_WEL) != 0 ? QSPI_STATUS_OK : QSPI_STATUS_ERROR);
+}
+
+qspi_status_t write_disable(Qspi &qspi)
+{
+    uint8_t reg[QSPI_STATUS_REGISTER_SIZE];
+    qspi.cmd.build(QSPI_CMD_WRDI);
+
+    if (qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    memset(reg, 0, QSPI_STATUS_REGISTER_SIZE);
+    if (read_register(STATUS_REG, reg, QSPI_STATUS_REGISTER_SIZE, qspi) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+
+    return ((reg[0] & STATUS_BIT_WEL) == 0 ? QSPI_STATUS_OK : QSPI_STATUS_ERROR);
+}
+
+WEAK void log_security_register(Qspi &qspi)
```

Comment:
What's the purpose of these log_xx_register functions? Why are they weak? Usually we don't have to print lot of data from tests, may be you need to confine them to debug builds only.

Reply:
Yes, it was for debugging purpose, I'll remove it